save-preprocessor/src/test/kotlin/org/cqfn/save/preprocessor/DownloadProjectTest.kt

```diff
@@ -0,0 +1,44 @@
+package org.cqfn.save.preprocessor
+
+import org.cqfn.save.repository.GitRepository
+
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.jupiter.api.Test
+import org.junit.runner.RunWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.web.client.TestRestTemplate
+import org.springframework.boot.test.web.client.postForEntity
+import org.springframework.http.HttpEntity
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.test.context.junit4.SpringRunner
+import java.io.File
+
+@RunWith(SpringRunner::class)
+@SpringBootTest(classes = [SaveApplication::class],
+        webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class DownloadProjectTest {
+
+    @Autowired
+    lateinit var testRestTemplate: TestRestTemplate
```

Comment:
For webflix usually WebClient (`WebTestClient` for tests) is preferred over rest template